src/components/ContactCard/ContactFieldInput.jsx

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Field } from "react-final-form";
+import classnames from "classnames";
+
+const getInputComponent = inputType =>
+  inputType === "textarea" ? "textarea" : "input";
+
+class ContactFieldInput extends React.Component {
+  state = {
+    renderLabel: false,
+    hasFocus: false
+  };
+
+  onFocus = () => {
+    this.setState({
+      renderLabel: this.props.withLabel,
+      hasFocus: true
+    });
+  };
+
+  onMainInputBlur = e => {
+    this.setState({
+      renderLabel: e.target.value && this.props.withLabel,
+      hasFocus: false
+    });
+  };
+
+  onSecondaryInputBlur = () => {
+    this.setState({
+      hasFocus: false
+    });
+  };
```

Comment:
I don't know what this PR aims to do. But at first sight, this component is a bit too complex. I can conclude on that because of these two functions `onMainInputBlur` and `onSecondaryInputBlur`. I keep trying to understand what the `ContactFieldInput` component is responsible for.